### PR TITLE
added Geo spatial Query

### PR DESCRIPTION
**Pull request by yousefhany77**
## Geospatial Query Operators
- [x]  Add typings for **`$geoIntersects`**
- [x]  Add typings for **`$near`**
- [x]  Add typings for **`$nearSphere`**
- [x]  Add typings for **`$geoWithin`**
    - - [x]  Add typings for **`$box`**
    - - [x]  Add typings for **`$center`**
    - - [x]  Add typings for **`$centerSphere`**
    - - [x]  Add typings for **`$polygon`**
- [x]  Add typings for **`$geometry`**
- [x]  Add typings for **`$maxDistance`**
- [x]  Add typings for **`$minDistance`**
- [x]  Add typings for **`$uniqueDocs`**
- [x]  Create assertion file for geo-spatial query types

### CODE REVIEW

File: src/types/geometry.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+export type Coordinates = [number, number];
+
+interface Geometry {
```
Comment (tianhuil) [inline src/types/geometry.ts:3]: Can you put a link to where these definitions come from?

File: src/types/geoSpatial.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+ * };
+ * ```
+ */
+
+export type GeoSpatialQueryWithSingleOption = (
+  | {
+      $geoIntersects: {
+        $geometry: GeoJSONPolygon;
+      };
+      $maxDistance?: never;
```
Comment (tianhuil) [inline src/types/geoSpatial.ts:97]: Why do you have `never` here?

File: src/types/geoSpatial.assert.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
+import * as ta from "type-assertions";
+import { GeoSpatialQueryWithSingleOption } from "./geoSpatial";
+import { type } from "os";
+
+type geoIntersectsExample = {
```
Comment (tianhuil) [inline src/types/geoSpatial.assert.ts:5]: Can you document where these types come from?

**Review by tianhuil — CHANGES_REQUESTED** [review]
This doesn't seem to be connected to the rest of the code.

File: src/types/geometry.ts (hunk 1 above)
Comment (yousefhany77) [reply src/types/geometry.ts:3]: Thank you for reviewing my pull request. I have added types to the GEOJSON objects used in the geo-spatial queries. 
I sourced these definitions from the official MongoDB documentation, which can be found here: https://www.mongodb.com/docs/manual/reference/geojson/#std-label-geospatial-indexes-store-geojson. I hope this clarifies the changes I have made

File: src/types/geoSpatial.ts (hunk 2 above)
Comment (yousefhany77) [reply src/types/geoSpatial.ts:97]: Regarding the use of never in the $maxDistance property of $geoIntersects: MongoDB's documentation specifies that the $maxDistance option can only be used with $near and $nearSphere, not with $geoIntersects.
 By setting the type of $maxDistance to never, we're ensuring that this option cannot be used in this context, thus conforming to MongoDB's rules and restrictions for geo-spatial queries."

File: src/types/geoSpatial.assert.ts (hunk 3 above)
Comment (yousefhany77) [reply src/types/geoSpatial.assert.ts:5]: Thank you for reviewing the pull request. I'm sorry, but I'm not sure which types you're referring to that I need to document. Could you please clarify that? Also, regarding the import statement for type from "os", I apologize for that mistake and I will remove it from the code.

Comment (tianhuil) [reply src/types/geoSpatial.assert.ts:5]: You should link to the documentation where you get the examples from.

File: src/types/geoSpatial.ts (hunk 2 above)
Comment (tianhuil) [reply src/types/geoSpatial.ts:97]: Why not just leave it out altogether?

**Comment by tianhuil** [comment]
1. So how do these types connect to the rest of the repo?  They aren't imported anywhere.
2. Please follow the conventions used in the code -- for example mongo's `Filter` type becomes `TsFilter`.

**Comment by tianhuil** [comment]
Thanks for your response.  I realize I actually have a hard time reading the diff so I added prettier.  Can you rebase and run prettier on this?

**Comment by yousefhany77** [comment]
Done rebased and ran prettier, but I don't know why all those files included in the changes

**Review by yousefhany77 — COMMENTED** [review]
I cleanup the PR from the changes i didn't make

**Review by tianhuil — APPROVED** [review]

**Pull request merged**